GliaWidgets/Sources/QueuesMonitor/QueuesMonitor.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import GliaCoreSDK
+import Combine
+
+class QueuesMonitor {
+    enum State {
+        case idle
+        case updated([Queue])
+        case failed(Error)
+    }
+
+    @Published private(set) var state: State = .idle
+
+    private let environment: Environment
+    private var subscriptionId: String?
+    private var queues: [Queue] = []
+
+    init(environment: Environment) {
+        self.environment = environment
+
+        startMonitoring()
+    }
+
+    func startMonitoring() {
```

Comment:
I think this logic can be covered with unit tests.

Reply:
Since unit tests will be added in further PRs, I'd add some TODO